### PR TITLE
fix(github): Ignore vulnerability alerts in FIXED or DISMISSED states

### PR DESCRIPTION
**Pull request by nyg**
## Changes

When querying Github's GraphQL API to retrieve vulnerability alerts, only retrieve those that are in OPEN state.

## Context

Closes #14316

The GraphQL request to retrieve vulnerabilities currently ignores their state (OPEN, FIXED or DISMISSED).

Fixed or dismissed vulnerabilities should not be considered as renovate will try to create PRs for these vulnerabilities, when actually the dependencies are already fixed (or the owner has decided they should not be fixed). This will also propose the PRs in the Dashboard issue under "Other branches" or "Ignored or blocked".

It seems this state was added recently, which may explain why this issue was not noticed before.

https://docs.github.com/en/graphql/reference/enums#repositoryvulnerabilityalertstate
https://docs.github.com/en/graphql/overview/changelog (Enum added on Feb 16th, 2022)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

*Note*: I have not written any test for this as this is a feature of Github's GraphQL API. Let me know if I should still be writing a test for it.

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [X] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
This will potentially break GHE and needs tests.

**Comment by rarkins** [comment]
If it's too complicated to add GHE version detection to this, it would be better to undo this change and add the appropriate filtering on our client side instead, so it works with all.

**Comment by viceice** [comment]
We already know if we on GHE, so check should be easy. We can enable filter on GHE when we know on which GHE version it's available later.

**Comment by rarkins** [comment]
But the problem this is addressing will still exist on GHE in the meantime?

**Comment by nyg** [comment]
> If it's too complicated to add GHE version detection to this, it would be better to undo this change and add the appropriate filtering on our client side instead, so it works with all.

Indeed, I think it would be a bit complicated to check for the GHE versions that support of not the new `state` field.

But then, how would you filter on the client side? Because if we add the `state` field to the query, then it's also going to fail with the GHE versions that don't support it.

One option I thought about, but don't know if it's doable, is schema introspection, i.e. to check if the `state` field is present or not, and depending on the result include or not the state field in the query.

> But the problem this is addressing will still exist on GHE in the meantime?

From what I understand, yes.

**Comment by rarkins** [comment]
Replaced by #14795 (perform ignore on the client side so that it works for GHE too)

**Comment by rarkins** [comment]
Reopening this, as I changed my mind :)

File: lib/modules/platform/github/index.ts [hunk 1]
```diff
@@ -1686,10 +1686,18 @@ export function massageMarkdown(input: string): string {
 
 export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   let vulnerabilityAlerts: { node: VulnerabilityAlert }[];
+
+  const gheSupportsStateFilter = semver.satisfies(
+    platformConfig.gheVersion,
+    '~3.0.25 || ~3.1.17 || ~3.2.9 || ~3.3.4'
```
[the remark this answers is not in the document]
Comment (nyg) [reply lib/modules/platform/github/index.ts]: Yes, I just pushed the change!

**Review by rarkins — DISMISSED** [review]

File: lib/modules/platform/github/index.spec.ts [hunk 2]
```diff
@@ -2462,6 +2462,57 @@ describe('modules/platform/github/index', () => {
...
+        });
+
+      await github.initPlatform({
+        endpoint: gheApiHost,
+        token: '123test',
+      });
+
+      const res = await github.getVulnerabilityAlerts();
+      expect(res).toHaveLength(1);
+      expect(httpMock.getTrace()).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/platform/github/index.spec.ts:2514]: Please remove this. Is obsolete

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

**Review by rarkins — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
Needs snapshot update

**Pull request merged**